src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/azuredevops/AzureDevOpsServerPullRequestDecorator.java

```diff
@@ -0,0 +1,385 @@
+        } catch (Exception ex) {
+            throw new IllegalStateException("Could not decorate Pull Request on AzureDevOps Server", ex);
+        }
+    }
+
+    private void handleIssue(AnalysisDetails analysisDetails, AzurePullRequestDetails azurePullRequestDetails,
+                             PostAnalysisIssueVisitor.ComponentIssue issue, ArrayList<CommentThread> azureCommentThreads) {
+        String filePath = analysisDetails.getSCMPathForIssue(issue).orElse(null);   
+
+        if (filePath != null) {
+            try {
+                Integer line = issue.getIssue().getLine();
+                filePath = filePath.endsWith("/") ? filePath : "/" + filePath;
+                LOGGER.trace(String.format("ISSUE: authorLogin: %s ", issue.getIssue().authorLogin()));
+                LOGGER.trace(String.format("ISSUE: key: %s ", issue.getIssue().key()));
+                LOGGER.trace(String.format("ISSUE: type: %s ", issue.getIssue().type().toString()));
+                LOGGER.trace(String.format("ISSUE: severity: %s ", issue.getIssue().severity()));
+                LOGGER.trace(String.format("ISSUE: componentKey: %s ", issue.getIssue().componentKey()));
+                LOGGER.trace(String.format("ISSUE: getLocations: %s ", Objects.requireNonNull(issue.getIssue().getLocations()).toString()));
+                LOGGER.trace(String.format("ISSUE: getRuleKey: %s ", issue.getIssue().getRuleKey()));
+                LOGGER.trace(String.format("COMPONENT: getDescription: %s ", issue.getComponent().getDescription()));
+                DbIssues.Locations locate = Objects.requireNonNull(issue.getIssue().getLocations());
+                boolean threadExists = false;
+                for (CommentThread azureThread : azureCommentThreads) {
+                    LOGGER.trace(String.format("azureFilePath: %s", azureThread.getThreadContext().getFilePath()));
+                    LOGGER.trace(String.format("filePath: %s (%s)", filePath, azureThread.getThreadContext().getFilePath().equals(filePath)));
+                    LOGGER.trace(String.format("azureLine: %d", azureThread.getThreadContext().getRightFileStart().getLine()));
+                    LOGGER.trace(String.format("line: %d (%s)", line, azureThread.getThreadContext().getRightFileStart().getLine() == locate.getTextRange().getEndLine()));
+
+                    // Check if thread already exists and close thread if issue is closed
+                    if (checkAzureThread(issue, filePath, azureThread, azurePullRequestDetails)) {
+                        threadExists = true;
+                        break;
+                    }
+                }
+                if (!issue.getIssue().getStatus().equals(Issue.STATUS_OPEN)) {
+                    LOGGER.info(String.format("SKIPPED ISSUE: Issue status is %s", issue.getIssue().getStatus()));
+                    return;
+                }
+
+                if (threadExists || !issue.getIssue().getStatus().equals(Issue.STATUS_OPEN)) {
```

Comment:
The condition `!issue.getIssue().getStatus().equals(Issue.STATUS_OPEN)` is already handled by previous if.

Reply:
Maybe you should test in case of threadExists if the thread is closed in azure and if so then reopen it?

Reply:
Removed the duplicate condition and added some logic to re-open Azure Threads if the SQ issue is open.

Reply:
Great. Thank you.

Reply:
After some thought on this, I think its best not to reactivate Azure threads if the issue is open in SQ.  Someone must have had to close that Azure thread on purpose with some status (wontFix, byDesign, resolved, etc...). As long as it was reported the first time, I don't think purposely closed threads need to be reopened.